data/src/main/java/com/microsoft/azure/kusto/data/format/CslFormat.java

```diff
@@ -0,0 +1,38 @@
+package com.microsoft.azure.kusto.data.format;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public abstract class CslFormat {
+    public abstract String getType();
+
+    public abstract Object getValue();
+
+    // Value must not be null
+    abstract String getValueAsString();
+
+    private String getValueOrNullAsString() {
+        if (getValue() == null) {
+            return "null";
+        } else {
+            return getValueAsString();
+        }
+    }
+
+    public String toString() {
+        return getType() + "(" + getValueOrNullAsString() + ")";
+    }
+
+    // For example, parses "int(7)" as "7"
+    // If type wasn't found in valueWithType, returns it unchanged
+    public static String parseValueFromValueWithType(String valueWithType, String type) {
+        String valueWithTypeRegex = String.format("%s\\s*\\(\\s*(.*\\S+)\\s*\\)\\s*", type);
+        Pattern valueWithTypePattern = Pattern.compile(valueWithTypeRegex, Pattern.CASE_INSENSITIVE);
+        Matcher matcher = valueWithTypePattern.matcher(valueWithType);
+        if (matcher.matches()) {
+            return matcher.group(1);
+        } else {

```

Comment:
I know this is unpopular in the "elegant" circles, but I personally consider this to be more readable.

Reply:
Do you mean the "no new line before else" part? 
Because that's fine in java

Reply:
I think the popular eleagent approach is:
`if (matcher.matches()) {
return matcher.group(1);
}
return valueWithType;`

Reply:
oh, then I agree with the elegant approach. (with new lines)

Reply:
I'm not a fan of its readability.

Reply:
Used ternary, as discussed